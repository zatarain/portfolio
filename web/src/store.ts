import { configureStore, ThunkAction, Action, createSlice } from '@reduxjs/toolkit'

export function makeStore() {
  const responsiveFlipper = createSlice({
    name: 'flipper',
    initialState: {
      responsive: false,
    },
    reducers: {
      flipResponsive: (state) => {
        state.responsive = !state.responsive
      }
    }
  })
  return configureStore({
    reducer: { flipper: responsiveFlipper.reducer },
  })
}

const store = makeStore()

export type AppState = ReturnType<typeof store.getState>

export type AppDispatch = typeof store.dispatch

export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  AppState,
  unknown,
  Action<string>
>

export default store
