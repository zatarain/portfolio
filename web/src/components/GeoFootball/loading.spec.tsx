import { Provider } from 'react-redux'
import store from '#store'
import { render } from '@testing-library/react'
import Loading from './loading'

describe('<Loading />', () => {
	it('renders the component correctly with passed name', () => {

		const { asFragment } = render(
			<Provider store={store}>
				<Loading text='Hello World' />
			</Provider>
		)

		expect(asFragment()).toMatchSnapshot()
	})
})
