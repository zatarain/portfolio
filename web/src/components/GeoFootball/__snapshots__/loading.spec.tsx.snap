// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> renders the component correctly with passed name 1`] = `
<DocumentFragment>
  <img
    alt="Hello World"
    data-nimg="1"
    decoding="async"
    height="16"
    loading="lazy"
    src="/_next/image?url=https%3A%2F%2Fi.gifer.com%2FVAyR.gif&w=32&q=75"
    srcset="/_next/image?url=https%3A%2F%2Fi.gifer.com%2FVAyR.gif&w=16&q=75 1x, /_next/image?url=https%3A%2F%2Fi.gifer.com%2FVAyR.gif&w=32&q=75 2x"
    style="color: transparent;"
    width="16"
  />
   Hello World
</DocumentFragment>
`;
